app/models/person.rb

```diff
@@ -365,6 +365,16 @@ def anonymize
     new_wca_id
   end
 
+  def private_attributes_for_user(user)
+    if user&.wca_id == wca_id || user&.any_kind_of_delegate?
+      %w[dob]
+    elsif user&.can_admin_results?
+      %w[incorrect_wca_id_claim_count dob]
+    else
+      []
+    end
```

Comment:
I think you can avoid a whole bunch of `&.` safe-navigation by just checking immediately if the user is nil: `return [] if user.nil?`

Reply:
Okay changed.